charms/worker/k8s/charmcraft.yaml

```diff
@@ -189,6 +189,54 @@ config:
       default: false
       description: |
         Enable/Disable the gateway feature on the cluster.
+    load-balancer-enabled:
+      type: boolean
+      default: false
+      description: |
+        Enable/Disable the load balancer feature on the cluster.
+    load-balancer-cidrs:
+      type: string
+      default: ""
+      description: |
+        Space-separated list of CIDRs to use for the load balancer. This is
+        only used if load-balancer-enabled is set to true.
+    load-balancer-l2-mode:
+      type: boolean
+      default: false
+      description: |
+        Enable/Disable L2 mode for the load balancer. This is only used if
+        load-balancer-enabled is set to true.
+    load-balancer-l2-interfaces:
+      type: string
+      default: ""
+      description: |
+        Space-separated list of interfaces to use for the load balancer. This
+        is only used if load-balancer-l2-mode is set to true. if unset, all
+        interfaces will be used.
```

Comment:
Something feels odd here. I suppose it's likely that all the interfaces used for the L2 load balancer will be the same on all the machines. But this could be better if it was a regex rather than a fixed list of interfaces not really known until the machine is deployed

Reply:
This is valid, it could be a feature we add in the future. I have no idea if MetalLB supports regex for the interfaces.

See: https://metallb.universe.tf/configuration/_advanced_l2_configuration/index.html#specify-network-interfaces-that-lb-ip-can-be-announced-from

Reply:
So, we could do the regex IN the charm.  It doesn't necessarily have to apply to metallb.  We've got to remain generic here

Reply:
Lets leave this regex idea as a future feature